bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ResumingStreamingResultScanner.java

```diff
@@ -93,52 +97,62 @@ public Row next() throws IOException {
         Row result = currentDelegate.next();
         if (result != null) {
           lastRowKey = result.getKey();
-          rowCount++;
-          // We've had at least one successful RPC, reset the backoff
-          currentBackoff = null;
+          rowCount.incrementAndGet();
+          // We've had at least one successful RPC, reset the backoff and retry counter
+          currentErrorBackoff = null;
+          timeoutRetryCount.set(0);
```

Comment:
timeoutRetryCount = null is a few nanoseconds faster for each call to .next().  I think that given performance considerations, we should also go back to long rowCount instead of AtomicLong.